cost-analyzer/templates/diagnostics-deployment.yaml

```diff
@@ -1,5 +1,5 @@
 {{- if .Values.diagnostics.enabled }}
-{{- if or .Values.global.thanos.enabled (not (empty .Values.kubecostModel.federatedStorageConfigSecret )) -}}
+{{- if or ( not (empty .Values.prometheus.server.extraSecretMounts )) (not (empty .Values.kubecostModel.federatedStorageConfigSecret )) -}}
```

Comment:
I don't fully understand this change. Are we assuming that when users are running Thanos, they will always have the `.Values.prometheus.server.extraSecretMounts` set? If so we should leave a comment here.

Reply:
That is correct.

No doubt there's a better way to do this. Probably check if the list of sidecarContainers has one with an image containing thanos?

Though I'd also consider forcing thanos configs to add a flag to enable diagnostics.

I was just following the logic of the federated store. 

Here are the values for a thanos agent:

```yaml
prometheus:
  server:
    enableAdminApi: true
    extraArgs:
      storage.tsdb.max-block-duration: 2h
      storage.tsdb.min-block-duration: 2h
      storage.tsdb.retention: 2w
    extraVolumes:
    - name: object-store-volume
      secret:
        secretName: kubecost-thanos
    global:
      external_labels:
        cluster_id: kc-demo-dev
    sidecarContainers:
    - args:
      - sidecar
      - --log.level=debug
      - --tsdb.path=/data/
      - --prometheus.url=http://127.0.0.1:9090
      - --objstore.config-file=/etc/config/object-store.yaml
      - --min-time=-3h
      image: thanosio/thanos:v0.29.0
      name: thanos-sidecar
```

Reply:
Gotcha! Was there any issue with `.Values.global.thanos.enabled` though?

Reply:
yes, it isn't required for agents, so they wouldn't ship diagnostics by default.
https://github.com/kubecost/poc-common-configurations/blob/main/aws/values-amazon-secondary.yaml